tgui/packages/tgui/interfaces/DestructiveAnalyzer.tsx

```diff
@@ -0,0 +1,134 @@
+import { useBackend } from '../backend';
+import { Button, Box, Section, NoticeBox } from '../components';
+import { Window } from '../layouts';
+
+type Data = {
+  server_connected: boolean;
+  loaded_item: string;
+  item_icon: string;
+  indestructible: boolean;
+  already_deconstructed: boolean;
+  recoverable_points: string;
+  node_data: NodeData[];
+  research_point_id: string;
+};
+
+type NodeData = {
+  node_name: string;
+  node_id: string;
+  node_hidden: boolean;
+};
+
+export const DestructiveAnalyzer = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const {
+    server_connected,
+    indestructible,
+    loaded_item,
+    item_icon,
+    already_deconstructed,
+    recoverable_points,
+    research_point_id,
+    node_data,
```

Comment:
```suggestion
    node_data = [],
```
Just prefer that arrays from the backend are either casted (ideally) or using an optional chain before we use `.map`